### PR TITLE
[metricbeat] [gcp] add redis regions support

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Adds `redis` service region/regions support.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/metricbeat/module/gcp/constants.go [hunk 1]
```diff
@@ -84,6 +85,7 @@ const (
 	StorageResourceLabelLocation  = "resource.label.location"
 	CloudSQLResourceLabelRegion   = "resource.labels.region"
 	DataprocResourceLabelLocation = "resource.label.region"
+	RedisResourceLabelRegion      = "resource.label.region"
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/gcp/constants.go:88]: Should we change the one above to `DataprocResourceLabelRegion` to match the name or maybe use `ResourceLabelRegion` for both dataproc and redis?

Comment (endorama) [reply x-pack/metricbeat/module/gcp/constants.go:88]: Can we move forward with merging this PR and keep this discussion going, as is not directly related to the code changes in this PR?

What do you think about removing any reference to `Location` or `Region` from these constants? I don't think it adds much value and like in this case is prone to errors.

**Review by endorama — APPROVED** [review]

File: x-pack/metricbeat/module/gcp/constants.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/gcp/constants.go:88]: Im ok either way, keeping Location/Region/Zone or not, just want to be consistent here.

Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/gcp/constants.go:88]: https://github.com/elastic/beats/pull/33892 I just created this PR as a followup to the comment here so we don't forget to do it :)

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**